File: JpocCommonCore/Common/Enum/PublicEnum.vb
Imports System.ComponentModel

Public Class PublicEnum
    Public Enum eApplicationType
        Undefined
        WindowForm
        WebApplication
        WebService
    End Enum
    ''' <summary>
    ''' ユーザー種類
    ''' </summary>
    Public Enum eUserType
        ''' <summary>
        ''' 未定
        ''' </summary>
        Undefined = 0
        ''' <summary>
        ''' 内部ユーザー(エルゼビア)
        ''' </summary>
        INT = 1
        ''' <summary>
        ''' 個人ユーザー
        ''' </summary>
        IND = 2
        ''' <summary>
        ''' 施設ユーザー
        ''' </summary>
        INS = 3
    End Enum

    ''' <summary>
    ''' 実行環境
    ''' </summary>
    ''' <remarks>
    ''' AppSetingsのENV_[XXXX]の[XXXX]部
    ''' </remarks>
    Public Enum eRuntimeEnvironment
        ''' <summary>
        ''' LOCAL(開発)環境
        ''' </summary>
        LOCAL
        ''' <summary>
        ''' IT環境
        ''' </summary>
        IT
        ''' <summary>
        ''' DEMO環境
        ''' </summary>
        DEMO
        ''' <summary>
        ''' Preview環境
        ''' </summary>
        PREV
        ''' <summary>
        ''' 本番環境
        ''' </summary>
        PROD
        ''' <summary>
        ''' VM
        ''' </summary>
        VM
        ''' <summary>
        ''' DEV
        ''' </summary>
        DEV
        ''' <summary>
        ''' PRE
        ''' </summary>
        PRE
        ''' <summary>
        ''' TEST
        ''' </summary>
        TEST
        ''' <summary>
        ''' FOUND_UAT
        ''' </summary>
        FOUND_UAT
        ''' <summary>
        ''' FOUND_LOCAL
        ''' </summary>
        FOUND_LOCAL
        ''' <summary>
        ''' FOUND_PROD
        ''' </summary>
        FOUND_PROD
        ''' <summary>
        ''' ELIB
        ''' </summary>
        ELIB
        ''' <summary>
        ''' ENURSE
        ''' </summary>
        ENURSE
    End Enum

    ''' <summary>
    ''' ページアクセス権
    ''' </summary>
    ''' <remarks>
    ''' None=アクセス権なし
    ''' ReferenceOnly=読み取り専用
    ''' Editable=読み書き可能
    ''' </remarks>
    Public Enum eAccsessRight
        ''' <summary>
        ''' アクセス権なし
        ''' </summary>
        None = 0
        ''' <summary>
        ''' 読み取り専用
        ''' </summary>
        ReferenceOnly = 1
        ''' <summary>
        ''' 読み書き可能
        ''' </summary>
        Editable = 2
    End Enum

    ''' <summary>
    ''' 対象ページ
    ''' </summary>
    ''' <remarks>タブメニューインデックス</remarks>
    Public Enum eTabTargetPage
        ''' <summary>
        ''' 症状・疾患
        ''' </summary>
        Disease = 0
        ''' <summary>
        ''' 評価・治療例
        ''' </summary>
        Situation = 1
        ''' <summary>
        ''' 画像
        ''' </summary>
        ImageList = 2
        ''' <summary>
        ''' エビデンス・解説
        ''' </summary>
        Evidence = 3
        ''' <summary>
        ''' 詳細情報
        ''' </summary>
        Actions = 4
    End Enum

    ''' <summary>
    ''' DAOで使用するキー項目の種類
    ''' </summary>
    ''' <remarks>検索キー項目を示す</remarks>
    Public Enum eKeyValueType
        ''' <summary>
        ''' DiseaseID
        ''' </summary>
        DiseaseID = 0
        ''' <summary>
        ''' SituationID
        ''' </summary>
        SituationID = 1
        ''' <summary>
        ''' SituationOrderSetID
        ''' </summary>
        SituationOrderSetID = 2
        ''' <summary>
        ''' SituationOrderSetParentID
        ''' </summary>
        SituationOrderSetParentID = 3
        ''' <summary>
        ''' SituationOrderSetSampleID
        ''' </summary>
        SituationOrderSetSampleID = 4
        ''' <summary>
        ''' SituationOrderSetPatientProfileID
        ''' </summary>
        SituationOrderSetPatientProfileID = 5
        ''' <summary>
        ''' SrlID
        ''' </summary>
        SrlID = 6
        ''' <summary>
        ''' Name
        ''' </summary>
        Name = 7
        ''' <summary>
        ''' Code
        ''' </summary>
        Code = 8
    End Enum

    ''' <summary>
    ''' アップロード時の編集モード
    ''' </summary>
    ''' <remarks>新規か変更かを示す</remarks>
    Public Enum eUploadFileMode
        ''' <summary>
        ''' 新規
        ''' </summary>
        [New] = 0
        ''' <summary>
        ''' 更新
        ''' </summary>
        Edit = 1
    End Enum

    ''' <summary>
    ''' DiseaseActionType及びActionItemsのデータ種別
    ''' </summary>
    ''' <remarks>ヘッダかボディかを示す</remarks>
    Public Enum eActionDataType
        ''' <summary>
        ''' 未定
        ''' </summary>
        Undifined = -1
        ''' <summary>
        ''' ボディ部
        ''' </summary>
        Body = 0
        ''' <summary>
        ''' ヘッダー行
        ''' </summary>
        Header = 1
    End Enum

    ''' <summary>
    ''' ページ種類
    ''' </summary>
    ''' <remarks>表示する内容を示す</remarks>
    Public Enum ePageType
        ''' <summary>
        ''' 未定
        ''' </summary>
        Undifined = 0
        ''' <summary>
        ''' 疾患・症状
        ''' </summary>
        Disease = 1
        ''' <summary>
        ''' 評価・治療例
        ''' </summary>
        Situation = 2
        ''' <summary>
        ''' 画像
        ''' </summary>
        ImageList = 3
        ''' <summary>
        ''' エビデンス
        ''' </summary>
        Evidence = 4
        ''' <summary>
        ''' 詳細情報
        ''' </summary>
        Actions = 5
        ''' <summary>
        ''' 検査
        ''' </summary>
        LabTest = 6
        ''' <summary>
        ''' 薬剤
        ''' </summary>
        Drug = 7
        ''' <summary>
        ''' 白本
        ''' </summary>
        Shirobon = 8
        ''' <summary>
        ''' Added 2016/10/04
        ''' </summary>
        Handout = 9
    End Enum

    ''' <summary>
    ''' ジャーナル（リファレンス）ソース区分
    ''' </summary>
    ''' <remarks>ジャーナルが何に紐付いているかを示す</remarks>
    Public Enum eJournalType
        ''' <summary>
        ''' すべて
        ''' </summary>
        All = 0
        ''' <summary>
        ''' 画像
        ''' </summary>
        Image = 1
        ''' <summary>
        ''' エビデンス
        ''' </summary>
        Evidence = 2
        ''' <summary>
        ''' Step1
        ''' </summary>
        Step1 = 3
    End Enum

    ''' <summary>
    ''' ユーザーロール値
    ''' </summary>
    ''' <remarks>
    ''' ユーザの役割番号<br />
    ''' T_JP_Roleと一致していること
    ''' </remarks>
    Public Enum eRole
        ''' <summary>
        ''' 不定
        ''' </summary>
        Undefine = 0
        ''' <summary>
        ''' エルゼビア管理者
        ''' </summary>
        <Description("ElsevierAdmin")>
        ElsevierAdmin = 1
        ''' <summary>
        ''' 施設管理者
        ''' </summary>
        <Description("InstitutionAdmin")>
        InstitutionAdmin = 3
        ''' <summary>
        ''' IPアドレスログイン
        ''' </summary>
        <Description("InstitutionIP")>
        InstitutionIP = 4
        ''' <summary>
        ''' 施設ユーザー
        ''' </summary>
        <Description("InstitutionUser")>
        InstitutionUser = 5
        ''' <summary>
        ''' 個人ユーザー
        ''' </summary>
        <Description("Individual")>
        Individual = 6
        ''' <summary>
        ''' トライアルユーザー
        ''' </summary>
        <Description("Trial")>
        Trial = 7
        ''' <summary>
        ''' エルゼビア編集者
        ''' </summary>
        <Description("ElsevierEditor")>
        ElsevierEditor = 8
        ''' <summary>
        ''' エルゼビアユーザー管理
        ''' </summary>
        <Description("ElsevierManager")>
        ElsevierManager = 9
        ''' <summary>
        ''' エルゼビア担当者
        ''' </summary>
        <Description("ElsevierUser")>
        ElsevierUser = 10
        ''' <summary>
        ''' VM管理者
        ''' </summary>
        <Description("VMAdmin")>
        VMAdmin = 11
        ''' <summary>
        ''' VMユーザ(自動ログインユーザ)
        ''' </summary>
        <Description("VMUser")>
        VMUser = 12
        ''' <summary>
        ''' 営業アカウント
        ''' </summary>
        <Description("ElsevierSales")>
        ElsevierSales = 13
        ''' <summary>
        ''' テクニカルサポート
        ''' </summary>
        <Description("TechnicalSupport")>
        TechnicalSupport = 14
        ''' <summary>
        ''' DVD ユーザー
        ''' </summary>
        <Description("DvdUser")>
        DvdUser = 15
        ''' <summary>
        ''' 代理店ユーザー
        ''' </summary>
        <Description("Agency")>
        Agency = 16
        ''' <summary>
        ''' VMコンテンツ作業者
        ''' </summary>
        <Description("VMContents")>
        VMContents = 17
        ''' <summary>
        ''' InstitutionSelfUser 2016 04 20
        ''' </summary>
        <Description("InstitutionSelfUser")>
        InstitutionSelfUser = 19
        ''' <summary>
        ''' Tentative
        ''' </summary>
        <Description("Tentative")>
        Tentative = 20
        <Description("RegistrationID")>
        RegistrationID = 21
        <Description("Subscriber")>
        Subscriber = 22
        <Description("Entitlement Audit")>
        EntitlementAudit = 23
        <Description("Entitlement Support")>
        EntitlementSupport = 24
        <Description("JP Support")>
        JPSupport = 25
        <Description("Elsevier Marketing")>
        ElsevierMarketing = 26
        <Description("Individual Management")>
        IndividualManagement = 27
        <Description("Support Admin")>
        SupportAdmin = 28
        <Description("RegistrationIDForM3")>
        RegistrationIDForM3 = 29
        <Description("RegistrationIDForTrialM3")>
        RegistrationIDForTrialM3 = 30
        <Description("TrialSubscriber")>
        TrialSubscriber = 31
    End Enum

    ''' <summary>
    ''' ログイン種類
    ''' </summary>
    Public Enum eLoginType
        ''' <summary>
        ''' 未設定（未ログイン）
        ''' </summary>
        Undifined = 0
        ''' <summary>
        ''' パスワード
        ''' </summary>
        Password = 1
        ''' <summary>
        ''' IPアドレス
        ''' </summary>
        IpAddress = 2
    End Enum

    ''' <summary>
    ''' メッセージ種類
    ''' </summary>
    ''' <remarks>
    ''' Undifined=未定
    ''' Question=質問
    ''' Information=情報
    ''' Warning=警告
    ''' Error=業務論理エラー
    ''' Fatal=システムエラー
    ''' </remarks>
    Public Enum eMessageType
        Undifined = 0
        Question = 1
        Information = 2
        Warning = 3
        [Error] = 4
        Fatal = 5
    End Enum


    ''' <summary>
    ''' 画像へのリンクを作成する際の挙動
    ''' </summary>
    Public Enum eImageLinkBehavior
        ''' <summary>
        ''' アイコンも表示しリンクも設定
        ''' </summary>
        IconAndLink = 0
        ''' <summary>
        ''' アイコンは表示するがリンクは張らない
        ''' </summary>
        IconOnly = 1
        ''' <summary>
        ''' アイコン非表示
        ''' </summary>
        HideIcon = 2
    End Enum
    Public Enum eRtnCD
        Normal = 0
        LogicalError = 1
        FaitalError = 9
    End Enum

    Public Enum eJobResult
        Success = 0
        Warning = 1
        [Error] = 9
    End Enum

    Public Enum eImageType
        Undefined = 0
        RawImage = 1
        TbImage = 2
        AdImage = 3
        DpImage = 4
    End Enum

    Public Enum eStepEventType
        Message = 0
        StartEvent = 1
        EndEvent = 2
    End Enum

    ''' <summary>
    ''' デモタイプ
    ''' </summary>
    Public Enum eDemoType
        ''' <summary>
        ''' 未デモ
        ''' </summary>
        None = 0
        ''' <summary>
        ''' デモ１
        ''' </summary>
        DEMO = 1
    End Enum

    Public Enum eButtonAction
        Undifined = 0
        ShowForm = 1
        ShowFormAsDialog = 2
        CloseForm = 3
        UpdateDataBase = 4
        SelectDataBase = 5
        AnyTransaction = 6
    End Enum

    Public Enum eKeyDataType
        [Integer]
        [String]
        CsvString
    End Enum

    Public Enum eInputType
        Undefined
        Entry
        SearchCriteria
        MultiPurpose
    End Enum

    Public Enum eContentType
        Text
        Image
    End Enum

    Public Enum eMinMax
        MIN
        MAX
    End Enum

    Public Shared Function TryParseEnumByName(enumType As Type, enumName As String) As Object
        If [Enum].IsDefined(enumType, enumName) Then
            Try
                Return [Enum].Parse(enumType, enumName, False)
            Catch
            End Try
        End If
        Return Nothing
    End Function



End Class

File: JPOC_VM_DEMO/Common/PublicEnum.cs
using System.ComponentModel;
namespace JPOC_VM_DEMO.Common
{

    /// <summary>
    /// Contains all public enumerations for the JPOC system
    /// </summary>
    public static class PublicEnum
    {
        public enum eApplicationType
        {
            Undefined,
            WindowForm,
            WebApplication,
            WebService
        }

        /// <summary>
        /// User type enumeration
        /// </summary>
        public enum eUserType
        {
            /// <summary>
            /// Undefined
            /// </summary>
            Undefined = 0,

            /// <summary>
            /// Internal user (Elsevier)
            /// </summary>
            INT = 1,

            /// <summary>
            /// Individual user
            /// </summary>
            IND = 2,

            /// <summary>
            /// Institution user
            /// </summary>
            INS = 3
        }

        /// <summary>
        /// Runtime environment
        /// </summary>
        /// <remarks>
        /// ENV_[XXXX] part of AppSettings
        /// </remarks>
        public enum eRuntimeEnvironment
        {
            /// <summary>
            /// LOCAL (Development) environment
            /// </summary>
            LOCAL,

            /// <summary>
            /// IT environment
            /// </summary>
            IT,

            /// <summary>
            /// DEMO environment
            /// </summary>
            DEMO,

            /// <summary>
            /// Preview environment
            /// </summary>
            PREV,

            /// <summary>
            /// Production environment
            /// </summary>
            PROD,

            /// <summary>
            /// VM environment
            /// </summary>
            VM,

            /// <summary>
            /// Development environment
            /// </summary>
            DEV,

            /// <summary>
            /// Pre-production environment
            /// </summary>
            PRE,

            /// <summary>
            /// Test environment
            /// </summary>
            TEST,

            /// <summary>
            /// Foundation UAT environment
            /// </summary>
            FOUND_UAT,

            /// <summary>
            /// Foundation local environment
            /// </summary>
            FOUND_LOCAL,

            /// <summary>
            /// Foundation production environment
            /// </summary>
            FOUND_PROD,

            /// <summary>
            /// E-Library environment
            /// </summary>
            ELIB,

            /// <summary>
            /// E-Nurse environment
            /// </summary>
            ENURSE
        }

        public enum eRtnCD
        {
            /// <summary>
            /// Operation completed successfully
            /// </summary>
            Normal = 0,

            /// <summary>
            /// Business logic error occurred
            /// </summary>
            LogicalError = 1,

            /// <summary>
            /// Critical system error occurred
            /// </summary>
            FatalError = 9  // Fixed typo from 'Faital' to 'Fatal'
        }

        /// <summary>
        /// Page access rights
        /// </summary>
        /// <remarks>
        /// None = No access rights
        /// ReferenceOnly = Read-only
        /// Editable = Read/Write enabled
        /// </remarks>
        public enum AccessRight
        {
            /// <summary>
            /// No access rights
            /// </summary>
            None = 0,

            /// <summary>
            /// Read-only access
            /// </summary>
            ReferenceOnly = 1,

            /// <summary>
            /// Read/Write access
            /// </summary>
            Editable = 2
        }

        /// <summary>
        /// Target page
        /// </summary>
        /// <remarks>Tab menu index</remarks>
        public enum TabTargetPage
        {
            /// <summary>
            /// Disease/Symptoms
            /// </summary>
            Disease = 0,

            /// <summary>
            /// Evaluation/Treatment examples
            /// </summary>
            Situation = 1,

            /// <summary>
            /// Images
            /// </summary>
            ImageList = 2,

            /// <summary>
            /// Evidence/Description
            /// </summary>
            Evidence = 3,

            /// <summary>
            /// Detailed information
            /// </summary>
            Actions = 4
        }

        /// <summary>
        /// Key item types used in DAO
        /// </summary>
        /// <remarks>Indicates search key items</remarks>
        public enum KeyValueType
        {
            DiseaseID = 0,
            SituationID = 1,
            SituationOrderSetID = 2,
            SituationOrderSetParentID = 3,
            SituationOrderSetSampleID = 4,
            SituationOrderSetPatientProfileID = 5,
            SrlID = 6,
            Name = 7,
            Code = 8
        }

        /// <summary>
        /// Upload file edit mode
        /// </summary>
        /// <remarks>Indicates new or update</remarks>
        public enum UploadFileMode
        {
            /// <summary>
            /// New
            /// </summary>
            New = 0,

            /// <summary>
            /// Update
            /// </summary>
            Edit = 1
        }

        /// <summary>
        /// Disease action type and action items data type
        /// </summary>
        /// <remarks>Indicates header or body</remarks>
        public enum ActionDataType
        {
            /// <summary>
            /// Undefined
            /// </summary>
            Undefined = -1,

            /// <summary>
            /// Body part
            /// </summary>
            Body = 0,

            /// <summary>
            /// Header row
            /// </summary>
            Header = 1
        }

        /// <summary>
        /// Page type
        /// </summary>
        /// <remarks>Indicates display content</remarks>
        public enum PageType
        {
            Undefined = 0,
            Disease = 1,
            Situation = 2,
            ImageList = 3,
            Evidence = 4,
            Actions = 5,
            LabTest = 6,
            Drug = 7,
            Shirobon = 8,
            Handout = 9
        }

        /// <summary>
        /// Journal (Reference) source category
        /// </summary>
        /// <remarks>Indicates what the journal is linked to</remarks>
        public enum JournalType
        {
            All = 0,
            Image = 1,
            Evidence = 2,
            Step1 = 3
        }

        /// <summary>
        /// User role values
        /// </summary>
        /// <remarks>
        /// User role numbers
        /// Must match T_JP_Role table
        /// </remarks>
        public enum eRole
        {
            [Description("Undefined")]
            Undefined = 0,

            [Description("ElsevierAdmin")]
            ElsevierAdmin = 1,

            [Description("InstitutionAdmin")]
            InstitutionAdmin = 3,

            [Description("InstitutionIP")]
            InstitutionIP = 4,

            [Description("InstitutionUser")]
            InstitutionUser = 5,

            [Description("Individual")]
            Individual = 6,

            [Description("Trial")]
            Trial = 7,

            [Description("ElsevierEditor")]
            ElsevierEditor = 8,

            [Description("ElsevierManager")]
            ElsevierManager = 9,

            [Description("ElsevierUser")]
            ElsevierUser = 10,

            [Description("VMAdmin")]
            VMAdmin = 11,

            [Description("VMUser")]
            VMUser = 12,

            [Description("ElsevierSales")]
            ElsevierSales = 13,

            [Description("TechnicalSupport")]
            TechnicalSupport = 14,

            [Description("DvdUser")]
            DvdUser = 15,

            [Description("Agency")]
            Agency = 16,

            [Description("VMContents")]
            VMContents = 17,

            [Description("InstitutionSelfUser")]
            InstitutionSelfUser = 19,

            [Description("Tentative")]
            Tentative = 20,

            [Description("RegistrationID")]
            RegistrationID = 21,

            [Description("Subscriber")]
            Subscriber = 22,

            [Description("Entitlement Audit")]
            EntitlementAudit = 23,

            [Description("Entitlement Support")]
            EntitlementSupport = 24,

            [Description("JP Support")]
            JPSupport = 25,

            [Description("Elsevier Marketing")]
            ElsevierMarketing = 26,

            [Description("Individual Management")]
            IndividualManagement = 27,

            [Description("Support Admin")]
            SupportAdmin = 28,

            [Description("RegistrationIDForM3")]
            RegistrationIDForM3 = 29,

            [Description("RegistrationIDForTrialM3")]
            RegistrationIDForTrialM3 = 30,

            [Description("TrialSubscriber")]
            TrialSubscriber = 31
        }

        /// <summary>
        /// Login type
        /// </summary>
        public enum eLoginType
        {
            /// <summary>
            /// Undefined (Not logged in)
            /// </summary>
            Undefined = 0,

            /// <summary>
            /// Password
            /// </summary>
            Password = 1,

            /// <summary>
            /// IP Address
            /// </summary>
            IpAddress = 2
        }

        /// <summary>
        /// Message type
        /// </summary>
        public enum MessageType
        {
            Undefined = 0,
            Question = 1,
            Information = 2,
            Warning = 3,
            Error = 4,
            Fatal = 5
        }

        /// <summary>
        /// Image link behavior
        /// </summary>
        public enum ImageLinkBehavior
        {
            /// <summary>
            /// Show icon and set link
            /// </summary>
            IconAndLink = 0,

            /// <summary>
            /// Show icon but no link
            /// </summary>
            IconOnly = 1,

            /// <summary>
            /// Hide icon
            /// </summary>
            HideIcon = 2
        }

        public enum ReturnCode
        {
            Normal = 0,
            LogicalError = 1,
            FatalError = 9
        }

        public enum JobResult
        {
            Success = 0,
            Warning = 1,
            Error = 9
        }

        public enum ImageType
        {
            Undefined = 0,
            RawImage = 1,
            TbImage = 2,
            AdImage = 3,
            DpImage = 4
        }

        public enum StepEventType
        {
            Message = 0,
            StartEvent = 1,
            EndEvent = 2
        }

        /// <summary>
        /// Demo type
        /// </summary>
        public enum DemoType
        {
            /// <summary>
            /// No demo
            /// </summary>
            None = 0,

            /// <summary>
            /// Demo 1
            /// </summary>
            DEMO = 1
        }

        public enum ButtonAction
        {
            Undefined = 0,
            ShowForm = 1,
            ShowFormAsDialog = 2,
            CloseForm = 3,
            UpdateDatabase = 4,
            SelectDatabase = 5,
            AnyTransaction = 6
        }

        public enum KeyDataType
        {
            Integer,
            String,
            CsvString
        }

        public enum InputType
        {
            Undefined,
            Entry,
            SearchCriteria,
            MultiPurpose
        }

        public enum ContentType
        {
            Text,
            Image
        }

        public enum MinMax
        {
            MIN,
            MAX
        }

        /// <summary>
        /// Tries to parse an enum value by name
        /// </summary>
        public static object TryParseEnumByName(Type enumType, string enumName)
        {
            if (Enum.IsDefined(enumType, enumName))
            {
                try
                {
                    return Enum.Parse(enumType, enumName, false);
                }
                catch
                {
                    return null;
                }
            }
            return null;
        }
    }
}